### PR TITLE
Dev/run infection model

### PR DESCRIPTION
**Pull request by matteopariset**
Periodically update infection probability and store it on Firestore

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/location/LocationServiceTest.java [hunk 1]
```diff
@@ -116,8 +122,9 @@ public void obtainNullWithoutPermissions() throws Throwable {
     }
 
     @Test
-    public void canStopAggregator() {
+    public void canStopAggregator() throws Throwable {
         mActivityRule.getActivity().getService().onProviderDisabled(LocationManager.GPS_PROVIDER);
         // TODO: This test should check the effects of the operation
```
Comment (TheCrazyKing) [inline app/src/androidTest/java/ch/epfl/sdp/location/LocationServiceTest.java:127]: TODO : done ? or to do ?

File: app/src/main/java/ch/epfl/sdp/contamination/CachingDataSender.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
     String privateUserFolder = "privateUser/";
     String privateRecoveryCounter = "recoveryCounter";
 
+    int MAX_CACHE_ENTRY_AGE = InfectionAnalyst.UNINTENTIONAL_CONTAGION_TIME;
```
Comment (TheCrazyKing) [inline app/src/main/java/ch/epfl/sdp/contamination/CachingDataSender.java:23]: isn't this useless?

File: app/src/main/java/ch/epfl/sdp/location/LocationService.java [hunk 3]
```diff
@@ -36,41 +47,117 @@
...
     private static final int MIN_UP_INTERVAL_METERS = 5;
 
+    public static final String ALARM_GOES_OFF = "beeep!";
+
+    public static final String INFECTION_PROBABILITY_TAG = "infectionProbability";
+    public static final String INFECTION_STATUS_TAG = "infectionStatus";
+    public static final String LAST_UPDATED_TAG = "lastUpdated";
+
+    // TODO: This value should be set to several hours. It's now 2 minutes to allow for demo
+    private static long alarmDelayMillis = 120_000;
```
Comment (TheCrazyKing) [inline app/src/main/java/ch/epfl/sdp/location/LocationService.java:57]: What is this all about ?

**Review by TheCrazyKing — APPROVED** [review]
Ok if the tests for authentication pass and that authentication is still possible, which I suppose is the case, this PR is clean . Nice job. Please answer my questions on your code

**Review by AdrienRem — APPROVED** [review]
That's a huge amount of work, thank you !

I don't have a change request

File: app/src/main/java/ch/epfl/sdp/firestore/FirestoreInteractor.java [hunk 4]
```diff
@@ -71,10 +58,26 @@
     //////////////////////////////////////////////////////////////
 
     public static <T> CompletableFuture<T> taskToFuture(Task<T> task) {
-        CompletableFuture<T> future = new CompletableFuture<>();
-        task.addOnSuccessListener(value -> future.complete(value));
-        task.addOnFailureListener(ex -> future.completeExceptionally(ex));
-        return future;
+        /*
```
Comment (TheBlueHawk) [inline app/src/main/java/ch/epfl/sdp/firestore/FirestoreInteractor.java:61]: Very nice improvement, I apology for forgetting the onComplete case during the transition from listeners to futures

File: app/src/androidTest/java/ch/epfl/sdp/location/CarrierUpdatePersistenceTest.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    public final ActivityTestRule<DataExchangeActivity> mActivityRule = new ActivityTestRule<>(DataExchangeActivity.class);
+
+    private Carrier iAmBob = new Layman(HEALTHY);
+    private AtomicInteger sentinel;
+
+    private static String fakeUserID = "THIS_IS_A_FAKE_ID";
+
+    @BeforeClass
+    public static void mockUserId() {
+        // To not pollute application status, make AuthenticationManager return a mock UserID
```
Comment (TheBlueHawk) [inline app/src/androidTest/java/ch/epfl/sdp/location/CarrierUpdatePersistenceTest.java:49]: Nice

**Review by TheBlueHawk — APPROVED** [review]
Good job!

File: app/src/main/java/ch/epfl/sdp/contamination/CachingDataSender.java (hunk 2 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/contamination/CachingDataSender.java:23]: Nope, without it encapsulation would be violated. Even if right now it has the same value of UNINTENTIONAL_CONTAGION_TIME, the size of the cache is logically independent

File: app/src/main/java/ch/epfl/sdp/location/LocationService.java (hunk 3 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/location/LocationService.java:57]: alarmDelayMillis is the (minimum) delay between two successive updates of the infection model

File: app/src/androidTest/java/ch/epfl/sdp/location/LocationServiceTest.java (hunk 1 above)
Comment (matteopariset) [reply app/src/androidTest/java/ch/epfl/sdp/location/LocationServiceTest.java:127]: Well spotted. That test now just checks that there are no exceptions. It should be enhanced in the future

**Pull request merged**